plugins/slipping-away-users/frontend/public/javascripts/countly.models.js

```diff
@@ -1,30 +1,106 @@
+                    newQuery[propertyNameWithoutUp] = propertyValue;
+                });
             }
-        });
+            return newQuery;
+        }
     };
 
-    countlySlippingPlugin.getData = function() {
-        return _data;
+    countlySlippingAwayUsers.service = {
+        fetchSlippingAwayUsers: function(filters, callback) {
+            $.ajax({
+                type: "GET",
+                url: countlyCommon.API_URL + "/o/slipping",
+                data: {
+                    app_id: countlyCommon.ACTIVE_APP_ID,
+                    method: 'slipping',
+                    query: JSON.stringify(countlySlippingAwayUsers.helpers.buildFilters(filters))
+                },
+                success: function(json) {
+                    callback(null, json);
+                },
+                error: function(error) {
+                    callback(error, null);
+                }
+            });
+        }
     };
 
-}(window.countlySlippingPlugin = window.countlySlippingPlugin || {}, jQuery));
\ No newline at end of file
+    countlySlippingAwayUsers.getVuexModule = function() {
+        var getInitialState = function() {
+            return {
+                slippingAwayUsers: [],
+                slippingAwayUsersFilters: { query: {}, byVal: []},
+                isLoading: false,
+                hasError: false,
+                error: null,
+            };
+        };
+
+        var slippingAwayUsersActions = {
+            fetchSlippingAwayUsers: function(context) {
+                context.dispatch('onFetchInit');
+                countlySlippingAwayUsers.service.fetchSlippingAwayUsers(context.state.slippingAwayUsersFilters, function(error, response) {
+                    if (error) {
+                        context.dispatch('onFetchError', error);
+                    }
+                    else {
+                        context.commit('setSlippingAwayUsers', response);
+                        context.dispatch('onFetchSuccess');
+                    }
+                });
+            },
+            onFetchInit: function(context) {
+                context.commit('fetchInit');
+            },
+            onFetchError: function(context, error) {
+                context.commit('fetchError', error);
+            },
+            onFetchSuccess: function(context) {
+                context.commit('fetchSuccess');
+            },
+            onSetSlippingAwayUsersFilters: function(context, filters) {
+                context.commit('setSlippingAwayUsersFilters', filters);
+            }
+        };
+
+        var slippingAwayUsersMutations = {
+            setSlippingAwayUsers: function(state, value) {
+                state.slippingAwayUsers = value;
+            },
+            setSlippingAwayUsersFilters: function(state, value) {
+                state.slippingAwayUsersFilters = value;
+            },
+            fetchInit: function(state) {
```

Comment:
Handling {Init, Error, Success} cases is a good pattern but naming could be better. Another word for "fetch", for instance.

Reply:
you're right,fetch does not have any meaning in our case since we do not use the fetch() method to make requests. please let me know if you think replacing it with ''request'' is better or if you have other ideas.

Reply:
My point wasn't about using the fetch() API, but rather about the action (don't confuse with the Vuex actions) we describe. I mean the thing you do there is not fetching something, but mutating the state depending on the status of fetch you triggered. So, perhaps we should use a wording such as "setRequestStatus". You could also refactor the logic so that there is only one mutator function (e.g. setRequestStatus('loading', 'error', 'success')) and do state changes in it.

Reply:
I understand and in a hindsight I saw it too that there was a discrepancy between method name and what it was doing. Since, I have used fetch throughout the model, I will keep it then and instead add ''set'' as prefix for all the fetch methods, e.g. setFetchInit(), setFetchSuccess() and setFetchError() (I don't mind using a single function but then we will have to operate on a flag argument and it can easily become bloated).